src/type/matrix/utils/broadcast.js

```diff
@@ -0,0 +1,84 @@
+import { factory } from '../../../utils/factory.js'
+
+const name = 'broadcast'
+
+const dependancies = ['concat']
+
+export const createBroadcast = /* #__PURE__ */ factory(
+  name, dependancies,
+  ({ concat }) => {
+    /**
+   * Broadcasts two matrices, and return both in an array
+   * It checks if it's possible with broadcasting rules
+   *
+   * @param {Matrix}   A      First Matrix
+   * @param {Matrix}   B      Second Matrix
+   *
+   * @return {Matrix[]}      [ broadcastedA, broadcastedB ]
+   */
+    return (A, B) => {
+      let sizeA
+      let sizeB
+      const N = Math.max(A._size.length, B._size.length) // max number of dims
+      if (A._size.length === B._size.length) {
```

Comment:
I think this section to initialize `sizeA` and `sizeB` can be simpified to something like the following, since `_padLeft` will simply return the same output when the array already has the correct size (and you get a clone for free, though I think that is not necessary):

```js
const N = Math.max(A._size.length, B._size.length) // max number of dims
if (A._size.every((dim, i) => dim === B._size[i])) {
  // If matrices have the same size return them
  return [A, B]
}
const sizeA = _padLeft(A._size, N, 0)
const sizeB = _padLeft(B._size, N, 0)
```

Reply:
I was going back and forth with this. I was trying to reduce the number of actions and iterations it has to do, so padding only if needed (since only one of them might need padding). But since a clone is needed anyways, maybe it end up being about the same. I certainly prefer the sizes being constant. I will test it and get back.

Reply:
After a few tests I like it much more like your suggestion as it clones or pads as needed without conditionals.

Before cloning I was having an issue because when I used `_padLeft` it mutated `A._size` and that generated all sort of errors aftwerwards. I think there might be a smarter way to work around it but cloning allowed me to leave `A._size` as is